### DISCUSSION TITLE
Can't list columns through API

### DISCUSSION THREAD

**Discussion by dogboydog** [community/community#24394] [Apps, API and Webhooks]
<p>Hi, I’m a bit confused by this message. I’m getting “Must have admin rights to Repository”. I am the owner of the repository and project in question, and the token I am using has every checkbox checked (while debugging)</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/bdf519ac398714248a2dae4ce9ee2eea139c5a45.png" data-download-href="/uploads/short-url/r6rmKb4VlhqwHs715sWWReftb6t.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/9920963/181084430-41c86083-b8ab-4669-bc6d-1ef4dbd6b079.png" alt="image" data-base62-sha1="r6rmKb4VlhqwHs715sWWReftb6t" width="690" height="270" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/bdf519ac398714248a2dae4ce9ee2eea139c5a45_2_690x270.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/bdf519ac398714248a2dae4ce9ee2eea139c5a45_2_1035x405.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/b/bdf519ac398714248a2dae4ce9ee2eea139c5a45.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/bdf519ac398714248a2dae4ce9ee2eea139c5a45_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1332×523 182 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<pre><code class="lang-bash">+ github_api https://api.github.com/projects/1002604/columns
+ curl https://api.github.com/projects/1002604/columns -H 'Authorization: token xxxx' -H 'Accept: application/vnd.github.inertia-preview+json'
{
  "message": "Must have admin rights to Repository.",
  "documentation_url": "https://developer.github.com/v3/projects/columns/#list-project-columns"
}
</code></pre>
<p>However, I can view the details for a given column with the same token:</p>
<pre><code class="lang-auto">+ curl https://api.github.com/projects/columns/9510223 -H 'Authorization: token xxxxx' -H 'Accept: application/vnd.github.inertia-preview+json'
{
  "url": "https://api.github.com/projects/columns/9510223",
...
</code></pre>

**Comment by francisfuzz** [comment, marked as answer]
<aside class="quote no-group" data-username="dogboydog" data-post="1" data-topic="125658">
<div class="title">
<div class="quote-controls"></div>
 dogboydog:</div>
<blockquote>
<pre><code class="lang-auto">+ github_api https://api.github.com/projects/1002604/columns
+ curl https://api.github.com/projects/1002604/columns -H 'Authorization: token xxxx' -H 'Accept: application/vnd.github.inertia-preview+json'
{
  "message": "Must have admin rights to Repository.",
  "documentation_url": "https://developer.github.com/v3/projects/columns/#list-project-columns"
}
</code></pre>
</blockquote>
</aside>
<p>:wave: <a class="mention" href="/dogboydog">@dogboydog</a>: That message indicates that you don’t have admin rights to the project board with that <code>id</code> of value <code>1002604</code>. As a next step, could you please double check the <code>id</code> of that project board by <a href="https://docs.github.com/en/rest/reference/projects#list-repository-projects">listing the repository projects</a>? The response is a JSON array where each element represents a project board in the repository. Each element has an <code>id</code> value that you can verify against that <code>id</code>. If they don’t match up, I suggest making a new request with the updated <code>id</code>. Otherwise, I’d be curious to see the response (masking any sensitive or private information) so we can investigate further. :male_detective:</p>

**Comment by dogboydog** [comment]
<p>Hey this was totally my fault. I don’t even know where I got the project ID in the OP now. I thought I pasted the API url from the event payload in my workflow, but it doesn’t seem like that’s the case. I double checked the ID and mine is different and I’m able to list it. Long day I guess. Thank you!</p>
